core/config/project_settings.cpp

```diff
@@ -68,12 +68,27 @@ String ProjectSettings::get_resource_path() const {
 	return resource_path;
 }
 
-String ProjectSettings::get_safe_project_name() const {
-	String safe_name = OS::get_singleton()->get_safe_dir_name(get("application/config/name"));
-	if (safe_name.is_empty()) {
-		safe_name = "UnnamedProject";
+String ProjectSettings::get_csharp_project_name() const {
+	String name = get_setting_with_override("dotnet/project/assembly_name");
+	if (!name.is_empty()) {
+		return name;
 	}
-	return safe_name;
+	name = get_setting_with_override("application/config/name");
+	Vector<String> invalid_chars = Vector<String>({ //
```

Comment:
Empty comment exists to convince clang-format to let me keep the next comment where it is.